core/variant/reduz_struct_proposal.h

```diff
@@ -0,0 +1,255 @@
+
+#if 0
+
+// Implementation in Array
+
+// ContainerTypeValidate needs to be changed:
+
+struct ContainerTypeValidate {
+	Variant::Type type = Variant::NIL;
+	StringName class_name;
+	Ref<Script> script;
+	LocalVector<ContainerTypeValidate> struct_members; // Added this for structs, assignment from structs with same layout but different member names should be allowed (because it is likely too difficult to prevent)
+	const char *where = "container";
+};
+
+// ArrayPrivate needs to be changed:
+
+class ArrayPrivate {
+public:
+	SafeRefCount refcount;
+	Vector<Variant> array;
+	Variant *read_only = nullptr; // If enabled, a pointer is used to a temporary value that is used to return read-only values.
+	ContainerTypeValidate typed;
+
+	// Added struct stuff:
+	uint32_t struct_size = 0;
+	StringName * struct_member_names = nullptr;
+	bool struct_array = false;
+
+	_FORCE_INLINE_ bool is_struct() const {
+		return struct_size > 0;
+	}
+
+	_FORCE_INLINE_ bool is_struct_array() const {
+		return struct_size > 0;
+	}
+
+	_FORCE_INLINE_ int32_t find_member_index(const StringName& p_member) const {
+		for(uint32_t i = 0 ; i<struct_size ; i++) {
+			if (p_member == struct_member_names[i]) {
+				return (int32_t)i;
+			}
+		}
+
+		return -1;
+	}
+
+	_FORCE_INLINE_ bool validate_member(uint32_t p_index,const Variant& p_value) {
+		// needs to check with ContainerValidate, return true is valid
```

Comment:
I've implemented some methods in ContainerValidate, but I'm still not entirely sure what to put here.